lib/private/Files/Cache/Cache.php

```
@@ -505,6 +505,13 @@ public function moveFromCache(ICache $sourceCache, $sourcePath, $targetPath) {
 			list($sourceStorageId, $sourcePath) = $sourceCache->getMoveInfo($sourcePath);
 			list($targetStorageId, $targetPath) = $this->getMoveInfo($targetPath);
 
+			if (is_null($sourceStorageId) || $sourceStorageId === false) {
+				throw new \Exception('Invalid source storage id: ' . $sourceStorageId);
```

We might need to throw a better exception.